Simplenote/SharedStorageMigrator.swift

```diff
@@ -0,0 +1,93 @@
+import Foundation
+import CoreData
+
+@objc
+class SharedStorageMigrator: NSObject {
+    /// Database Migration
+    /// To be able to share data with app extensions, the CoreData database needs to be migrated to an app group
+    /// Must run before Simperium is setup
+
+    @objc
+    func performMigrationIfNeeded() {
+        // Confirm if the app group DB exists
+        guard mustPerformMigration else {
+            NSLog("Core Data Migration not required")
+            return
+        }
+
+        migrateCoreDataToAppGroup()
+    }
+
+    private var mustPerformMigration: Bool {
+        !CoreDataManager.appGroupDbExists && CoreDataManager.oldDbExists
```

Comment:
We should probably implement this logic here (and not to rely on `CoreDataManager`)